wagtail/admin/tests/tests.py

```diff
@@ -333,15 +333,36 @@ def test_no_permissions(self):
 class Test404(TestCase, WagtailTestUtils):
     def test_admin_404_template_used(self):
         self.login()
-        response = self.client.get('/admin/sdfgdsfgdsfgsdf')
+        response = self.client.get('/admin/sdfgdsfgdsfgsdf', follow=True)
+
+        # Check 400 error after CommonMiddleware redirect
         self.assertEqual(response.status_code, 404)
         self.assertTemplateUsed(response, 'wagtailadmin/404.html')
 
     def test_not_logged_in_redirect(self):
-        response = self.client.get('/admin/sdfgdsfgdsfgsdf')
+        response = self.client.get('/admin/sdfgdsfgdsfgsdf/')
 
         # Check that the user was redirected to the login page and that next was set correctly
-        self.assertRedirects(response, reverse('wagtailadmin_login') + '?next=/admin/sdfgdsfgdsfgsdf')
+        self.assertRedirects(response, reverse('wagtailadmin_login') + '?next=/admin/sdfgdsfgdsfgsdf/')
+
+
+class TestAdminURLAppendSlash(TestCase, WagtailTestUtils):
+    def setUp(self):
+        # Find root page
+        self.root_page = Page.objects.get(id=2)
+
+    def test_return_correct_view_for_correct_url_without_ending_slash(self):
+        self.login()
+
+        # APPEND_SLASH is True by default
+        # Remove trailing slash from URL
+        response = self.client.get(reverse('wagtailadmin_explore_root')[:-1], follow=True)
```

Comment:
Suggestion: could we explicitly set `APPEND_SLASH`  instead of assuming that it is set to true? I feel like that would make this test more robust.

Overriding settings in Django's TestCase can be done using the `settings()` context manager. [See overriding settings in the Django docs](https://docs.djangoproject.com/en/3.2/topics/testing/tools/#overriding-settings).

It should look somewhat like this:

```python
with self.settings(APPEND_SLASH=True):
    response = self.client.get(reverse('wagtailadmin_explore_root')[:-1], follow=True)
```